CSS animations and transitions property parsing

ewilligers opened the pull request:
Test the parsing and serialization of the longhand properties from
CSS Animations and CSS Transitions.

Relevant browser bugs:

Edge animation-duration should reject negatives
https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/18347049/

Edge and Safari incorrectly accept unitless 0 on `<time>` values
https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/18348525/
https://bugs.webkit.org/show_bug.cgi?id=113230

ericwilligers committed:
CSS animations and transitions property parsing

Test the parsing and serialization of the longhand properties from
CSS Animations and CSS Transitions.

Relevant browser bugs:

Edge animation-duration should reject negatives
https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/18347049/

Edge and Safari incorrectly accept unitless 0 on <time> values
https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/18348525/
https://bugs.webkit.org/show_bug.cgi?id=113230

birtles commented on css/css-animations/parsing/animation-name-invalid.html:
```diff
@@ -0,0 +1,19 @@
+</head>
+<body>
+<script>
+test_invalid_value("animation-name", '12');
```
Can we add a test that 'none' is invalid. As per:
"The <custom-ident> additionally excludes the none keyword."
(https://drafts.csswg.org/css-animations/#keyframes)
And CSS-wide keywords like "inherit" and "default": As per:
"The CSS-wide keywords are not valid <custom-ident>s."
(https://drafts.csswg.org/css-values-4/#custom-idents)

ewilligers replied:
none is a valid value for the property, it simply isn't a keyword.

For CSS-wide keywords like inherit and initial, I'd prefer a separate test to check that all properties of a spec support them. Like testing inheritance/non-inheritance
https://github.com/web-platform-tests/wpt/pull/12593

birtles commented on css/css-animations/parsing/animation-name-valid.html:
```diff
@@ -0,0 +1,24 @@
+test_valid_value("animation-name", 'ease-in');
+test_valid_value("animation-name", 'infinite');
+test_valid_value("animation-name", 'paused');
+test_valid_value("animation-name", 'first, second, third');
```
Can we add some tests for "strings"?
And escaped spaces in custom idents?
And otherwise-forbidden CSS-wide keywords which are allowed as strings, e.g. "initial"?
You can see some examples here: https://hg.mozilla.org/mozilla-central/rev/f41a419197de

ewilligers replied:
Tests added. They pass in Firefox.

birtles commented on css/css-animations/parsing/animation-delay-valid.html:
```diff
@@ -0,0 +1,21 @@
+<html>
+<head>
+<meta charset="utf-8">
+<title>CSS Animations Level 1: parsing animation-delay with valid values</title>
```
We can probably drop "Level 1" here.

birtles commented on css/css-animations/parsing/animation-delay-valid.html:
```diff
@@ -0,0 +1,21 @@
+<head>
+<meta charset="utf-8">
+<title>CSS Animations Level 1: parsing animation-delay with valid values</title>
+<link rel="author" title="Eric Willigers" href="mailto:ericwilligers@chromium.org">
```
(For what it's worth, we've been dropping the author annotations in files we've been submitting since it just seems to grow over time, but if you think it's useful feel free to leave it. We've also been dropping `<html>` and `<head>` tags but that's up to you too.)

birtles reviewed (approved):
Looks good to me with the extra `animation-name` tests added.

ericwilligers committed:
Test animation-name strings

The CSS-keywords initial, inherit, unset are not valid animation names.
default is also not permitted as a custom identifier animation name.

animation names can be strings.

birtles commented:
This is great, thanks Eric!
